lib/createCookieEncryptor.js

```diff
@@ -0,0 +1,73 @@
+import base64url from 'base64url'
+
+const KEY_ALGORITHM = 'AES-GCM'
+
+const toHexString = bytes =>
+  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '')
+
+const fromHexString = hexString =>
+  new Uint8Array(hexString.match(/.{1,2}/g).map(byte => parseInt(byte, 16)))
+
+export const createCookieEncryptor = (cookieName, encryptionKey) => {
+  const getKey = async () => {
+    try {
+      return crypto.subtle.importKey(
+        'raw',
+        base64url.toBuffer(encryptionKey),
+        { name: KEY_ALGORITHM },
+        false,
+        ['encrypt', 'decrypt']
+      )
+    } catch (e) {
+      console.error({ getKeyError: e })
+    }
+  }
+
+  const encrypt = async str => {
+    try {
+      const key = await getKey(KEY_ALGORITHM)
+      const iv = crypto.getRandomValues(new Uint8Array(16))
+      const encoder = new TextEncoder()
+      const data = encoder.encode(str)
+      const encrypted = await crypto.subtle.encrypt({ name: KEY_ALGORITHM, iv }, key, data)
+      return `${toHexString(iv)}${base64url(encrypted)}`
+    } catch (e) {
+      console.error({ encryptError: e })
+    }
```

Comment:
We actually want to throw error here rather than simply log out a message. Lets throw an http error her as well, 

Probably 400 Bad Request (for both the encrypt and decrypt methods)